frappe/model/base_document.py

```diff
@@ -371,6 +371,14 @@ def get_valid_dict(
 				elif df.fieldtype in float_like_fields and not isinstance(value, float):
 					value = flt(value)
 
+				elif df.fieldtype == "Rating":
+					if not isinstance(value, float):
+						value = flt(value)
+					if value < 0:
+						value = 0
+					elif value > 1:
+						value = 1
```

Comment:
```suggestion
					value = max(0, min(1, value))
```

Is this any better?

Reply:
Yep makes sense, switched to this and added a comment explaining incase it doesn't feel readable for someone just quickly glancing through.